app/src/main/java/com/google/android/sambadocumentsprovider/provider/SambaDocumentsProvider.java

```diff
@@ -198,6 +198,11 @@ public Cursor queryRoots(String[] projection) throws FileNotFoundException {
       R.drawable.ic_cloud,
     });
 
+    int rootFlags = Root.FLAG_SUPPORTS_CREATE | Root.FLAG_SUPPORTS_IS_CHILD;
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
```

Comment:
No need to check this. The flag can't be used for anything else even in Android N.